### DISCUSSION TITLE
About c source and c-include-dirs

### DISCUSSION THREAD

**Discussion by LukyGuyLucky** [c3lang/c3c#1340] [Q&A]
Just noticed 'c sources,cc,c-include-dirs' in the main branch.May I ask does this mean a c3c project can also contain c sources and then c3c build will compile all c3c sources and c source (assume cc/gcc is available )?So one can write c codes in a c3c program together with c3c codes?

Thanks.

```
"sources": [ "src/**" ],
  // C sources if the project also compiles C sources
  // relative to the project file.
  // "c-sources": [ "csource/**" ],
  // Include directories for C sources relative to the project file.
  // "c-include-dirs: [ "csource/include" ],
  // Output location, relative to project file.
  "output": "../build",
  // Architecture and OS target.
  // You can use 'c3c --list-targets' to list all valid targets.
  // "target": "windows-x64",
  // Targets.
  "targets": {
    "complete": {
      // Executable or library.
      "type": "executable",
      // Additional libraries, sources
      // and overrides of global settings here.
    },
  },
  // Global settings.
  // C compiler if the project also compiles C sources
  // defaults to 'cc'.
  "cc": "cc",

```

**Comment by lerno** [comment]
You can have c3c invoke a c compiler and combine the output of the C code with the output from C3. It's a convenience. There is no C compiler bundled inside of c3c.

**Reply by LukyGuyLucky** [reply]
I am sorry but I don't quite understand.I also tried this but failed.: 
[https://github.com/louiidev/c_interop_with_c3](url)

Looking forward to a snippet in the examples folder in c3c package regarding c->c3 and also c3->c.

**Reply by lerno** [reply]
You can look at the testproject file in github, it compiles and integrates with some c sources.

**Comment by LukyGuyLucky** [comment]
That's fantastic!
It is very impressive!!
Thank you so much!

I have 2 more questions here:
1.For c function puts in the c file,if I changed to printf,it won't compile as the  compiler complaints 
```lld-link: error: undefined symbol: __mingw_vfprintf```
I am wondering that's why?
2.If I add
``` 
#include <stdio.h>
puts("from test.c"); 
```
into csources/test.c,
The project compiles and runs as usual,but "from test.c" won't show when running.that's why?

Appreciated.

**Reply by lerno** [reply]
I assume there is something more than that in the file? Some function where the `puts` is placed an you call that from your main C program?

**Reply by LukyGuyLucky** [reply]
I just added the above mentioned two lines to  csources/test.c,nothing else been touched.

**Reply by lerno** [reply]
You need to add the mingw libc libraries if you're using mingw libc. If you instead want to use MSVC add `--cc  cl.exe`, this should also solve the problem (if visual studio is installed)
